### PR TITLE
Add note on WarningsAsErrors

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Cc @ivdiazsa

### CODE REVIEW

File: docs/workflow/README.md [hunk 1]
```diff
@@ -87,3 +87,7 @@ And how to measure performance:
 
 * [Benchmarking workflow for dotnet/runtime repository](https://github.com/dotnet/performance/blob/master/docs/benchmarking-workflow-dotnet-runtime.md)
 * [Profiling workflow for dotnet/runtime repository](https://github.com/dotnet/performance/blob/master/docs/profiling-workflow-dotnet-runtime.md)
+
+## Warnings as Errors
+
+The repo build treats warnings as errors. Dealing with warnings when you're in the middle of making changes can be annoying (e.g. unused variable that you plan to use later). To disable treating warnings as errors, set the `WarningsAsErrors` environment variable to `false` before building. This variable will be respected by both the `build.sh`/`build.cmd` root build scripts and builds done with `dotnet build` or Visual Studio. Some people may prefer setting this environment variable globally in their machine settings.
```
Comment (am11) [inline docs/workflow/README.md:93]: Internal and public alike, environment variables naming convention in the repo is

DOTNET_TREAT_WARNING_AS_ERROR=false or =0

Comment (MichalStrehovsky) [reply docs/workflow/README.md:93]: This sets an MSBuild property that is documented to be named like this.

**Review by marek-safar — APPROVED** [review]

**Comment by ivdiazsa** [comment]
Thanks for adding this @MichalStrehovsky! The note looks very much on point.

File: docs/workflow/README.md (hunk 1 above)
Comment (jeffhandley) [reply docs/workflow/README.md:93]: I noticed the `TreatWarningsAsErrors` is checked here:
https://github.com/dotnet/runtime/pull/81402/files#diff-0eee3e18fdac60b29b08c403d130ff9badcadeb22e2e8f01b1ddd34fdeb59ef0R270

This confused me as to what environment variable name we should be using: `TreatWarningsAsErrors`, or `WarningsAsErrors`?

From what I understood (https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/errors-warnings), `TreatWarningsAsErrors` is intended to be a boolean, whereas `WarningsAsErrors` is the list of diagnostic ids to upgrade from warnings to errors.

Comment (MichalStrehovsky) [reply docs/workflow/README.md:93]: Sigh, yeah, it's TreatWarningsAsErrors. I'll fix it.

**Pull request merged**